monadic-rx/shared/src/main/scala/mhtml/rx.scala

```diff
@@ -250,6 +262,18 @@ object Rx {
         Cancelable { () => cc.cancel; self.imitating = false }
       } else run(other)(effect)
 
+    case Sharing(self) =>
+      if (!self.isSharing) {
+        self.sharingCancelable = self.impure.foreach{ x =>
+          self.sharingMemo = Option(x)
+          println(s"hello from sharing: $x") // DEBUG
+          effect(x)
+        }
+        self.isSharing = true
+      }
+      else self.sharingMemo.foreach(x => effect(x.asInstanceOf[A]))
```

Comment:
You should be able to `assert(self.sharingMemo.nonEmpty)` here

Reply:
Not sure I follow , why is that preferable to `Option.foreach`?

Reply:
`Option.foreach` will silently do nothing if they are no memoized values. IIUC you should know here, by design, than `sharingMemo` is never `None`

Reply:
Ah, right; I guess `Option.get` would serve both purposes, in that case. I suppose I was erring on the side of leaning on language safety features when I probably don't need to. Maybe that's still good practice generally, but probably not in library calls that are frequently used like here.